rc/extra/ghcid.kak

```diff
@@ -0,0 +1,129 @@
+# ghcid integration (loosely based on make.kak)
+#
+# Starts ghcid in the background. all commands are executed on the
+# client specified by the 'ghcidtoolsclient' option.
+#
+# Recommended way to use this is to define a command in your kakrc
+# to split off a new client for showing the ghcid output:
+#
+#   set-option global ghcidtoolsclient ghcid
+#   define-command ghcid %{
+#     iterm-new-vertical "rename-client ghcid; ghcid-start"
+#   %}
+#
+# And you may want to bind the next and previous error commands to nice keys,
+# e.g. here ",p", and ",n":
+#   map -docstring "jump to previous error" global user p ':ghcid-previous-error<ret>'
+#   map -docstring "jump to next error" global user n ':ghcid-next-error<ret>'
+
+# options
+
+declare-option -docstring "name of the client in which to jump" \
+  str ghcidclient
+
+declare-option -docstring "name of the client in which to process output" \
+  str ghcidtoolsclient
+
+declare-option -docstring "ghcid error regexp" \
+  str ghcid_error_pattern "[^:\n]+:\d+:\d+: error:$"
+
+declare-option -hidden int ghcid_current_error_line
+
+# commands
+
+define-command -params .. \
+  -docstring %{ghcid-start [<arguments>]: ghcid utility wrapper
+All optional arguments forwarded to ghcid} \
+  ghcid-start %{ %sh{
+    workdir=$(mktemp -d "${TMPDIR:-/tmp}"/kak-ghcid.XXXXXXXXX)
+    output=${workdir}/output
+    errors=${workdir}/errors
+    fifo=${workdir}/fifo
+    exec 2> ${errors}
+    touch ${output}
+    mkfifo ${fifo}
+    cmd="ghcid -S -n -o ${output} $@"
+    echo "starting: ${cmd}" > ${output}
+    $cmd > /dev/null < /dev/null &
+    pid=$!
+    (tail -qf ${errors} ${output} > ${fifo} 2>&1) > /dev/null 2>&1 < /dev/null &
```

Comment:
The `-q` flag is not POSIX, expansions should be enclosed in double quotes, and you could use braces `{`/`}` instead of parenthesis `(`/`)` to avoid spawning an entire process.

Reply:
Ah that's unfortunate as without "-q" the output is quite noisy. That option is supported by both GNU and BSD variants so IMO it would be fine here.

Reply:
I see that `-q` is fairly largely supported across the implementations I've checked, despite not being POSIX compliant.

If there's no other way to implement that bit, I suppose it's fine to leave the flag until somebody complains about it?

Reply:
I'm having some issues with having tail being executed from the main shell instead of subshell. Kakoune seems to get stuck no matter how stdin/stdout/stderr is being redirected. Tried also disowning the tail process. Suggestions welcome on what might be going wrong.